### DISCUSSION TITLE
My Select component isn't triggering onChange for the form

### DISCUSSION THREAD

**Discussion by chrissyast** [react-hook-form/react-hook-form#4043] [Q&A]
My plan is to have a generic `onChange` method which will be fired whenever _any_ part of the form is changed, so I have `<form onchange={foo}`.

In my example, the TextField hits this method but the Select doesn't. I've tried registering the component with useEffect as per [this comment](https://github.com/react-hook-form/react-hook-form/issues/497#issuecomment-553206030) but still I'm not getting anything.

My codesandbox is [here](https://codesandbox.io/s/young-silence-eocgd?file=/src/components/Login.tsx)

**Comment by bluebill1049** [comment]
The codesandbox is broken, I think you meant to use `Controller` for controlled components.

**Reply by chrissyast** [reply]
Sorry @bluebill1049 , I had unsaved changes in the CSB. The same link works now.

**Reply by chrissyast** [reply]
@bluebill1049 I am using the `Controller` but still nothing

**Reply by bluebill1049** [reply]
please read the doc and follow the example here: https://react-hook-form.com/api#Controller

https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r

controller doesn't need register, it does that for you.

**Reply by chrissyast** [reply]
@bluebill1049  I've read that doc about 20 times and still can't understand it. I don't see where the destructured props within the `render` method are coming from...in the example on the docs the `onChange` method just seems to appear but is never defined. I've done everything the same as in the example but still the `Select` doesn't hit the `form`'s onChange method.

On the CSB, if you add an onChange method to the `form`, none of the select-type components hit the method, nor sliders, text fields and radios do. See [this fork](https://codesandbox.io/s/react-hook-form-v6-controller-forked-tyl3g?file=/src/index.js) where I have implemented this and only this change.

**Reply by bluebill1049** [reply]
read my comment above...

why there is regster={register} in the controller, can you find any example we use that in the doc?
<img width="378" alt="Screen Shot 2021-02-03 at 9 08 53 am" src="https://user-images.githubusercontent.com/10513364/106669217-70e42d80-65ff-11eb-8cf0-667f3075e0e9.png">

**Reply by chrissyast** [reply]
@bluebill1049 https://codesandbox.io/s/hardcore-volhard-7klt4?file=/src/components/Login.tsx `ref={register}` is commented out, still same behaviour
